src/jsonpath/interpreter.c

```diff
@@ -107,25 +103,20 @@ void exec_wildcard(zval* arr_head, zval* arr_cur, struct ast_node* tok, zval* re
   ZEND_HASH_FOREACH_END();
 }
 
-void exec_recursive_descent(zval* arr_head, zval* arr_cur, struct ast_node* tok, zval* return_value) {
+static void exec_recursive_descent(zval* arr_head, zval* arr_cur, struct ast_node* tok, zval* return_value) {
   if (arr_cur == NULL || Z_TYPE_P(arr_cur) != IS_ARRAY) {
     return;
   }
 
-  zval* data;
-  zval* zv_dest;
-  zend_string* key;
-  zend_ulong num_key;
-
   eval_ast(arr_head, arr_cur, tok, return_value);
 
-  ZEND_HASH_FOREACH_KEY_VAL(HASH_OF(arr_cur), num_key, key, data) {
-    exec_recursive_descent(arr_head, data, tok, return_value);
-  }
+  zval* data;
+
+  ZEND_HASH_FOREACH_VAL_IND(HASH_OF(arr_cur), data) { exec_recursive_descent(arr_head, data, tok, return_value); }
```

Comment:
Formatting: on three lines would be more readable.

Reply:
This is done automatically by clang format. I'll look into updating the rules.